pages/gallery/gallery.js

```diff
@@ -0,0 +1,420 @@
+  methods: {
+    openModal(id) {
+      this.modalTextureID = id
+      this.modalOpen = true
+
+      axios.get('/gallery/dialog/' + id)
+        .then(res => {
+          this.modalTextureObj = res.data
+        })
+    },
+    closeModal() {
+      this.modalOpen = false
+      this.modalTextureID = null
+      this.modalTextureObj = {}
+    },
+    splittedTextures() {
+      const result = []
+      const length = Object.keys(this.displayed.textures).length
+
+      for (let i = 0; i < length; i++) {
+        result.push(Object.values(this.displayed.textures)[i])
+      }
+
+      return result
+    },
+    getAuthor(textureID) {
+      let contributionsHTML = ''
+
+      let contributions = this.displayed.contributions[this.current.resolution === '32x' ? 'c32' : 'c64'][textureID]
+      if (contributions) {
+        const timestamp = contributions.reduce((a, b) => a = a > b.date ? a : b.date, 0)
+        const contribution = contributions.filter(el => el.date == timestamp)[0]
+
+        contributionsHTML = `
+          <li>
+            <p><i class="icon-people${contribution.contributors.length == 1 ? '' : 's'}"></i>${contribution.contributors.map(c => this.discordIDtoName(c)).join(', ')}</p>
+            <p><i class="icon-time"></i>${this.timestampToDate(timestamp)}</p>
+          </li>
+        `
+      }
+      else contributionsHTML = `<li class="danger-text"><p>${this.$root.lang().gallery.error_message.contribution_not_found}</p></li>`
+
+      if (this.current.resolution === '16x') contributionsHTML = `<li><i class="icon-mojang-red"></i>Mojang Studio</li>`
+
+      return `
+      <div class="texture-tooltip">
+        <div class="texture-info-container">
+          <h1 align="left" class="encased">#${textureID}&nbsp;&dash;&nbsp;${this.displayed.textures[textureID].name}</h1>
+          <ul align="left" class="encased">${contributionsHTML}</ul>
+        </div>
+        <div class="texture-tags-container">
+          <span class="encased">#${this.displayed.textures[textureID].tags.join('</span><span class="encased">#')}</span>
+        </div>
+      </div>
+      `
+    },
+    discordIDtoName(d) {
+      return this.displayed.contributors[d].username || this.$root.lang().gallery.error_message.user_not_found
+    },
+    timestampToDate(t) {
+      const a = new Date(t)
+      const months = [this.$root.lang().global.months.jan, this.$root.lang().global.months.feb, this.$root.lang().global.months.mar, this.$root.lang().global.months.apr, this.$root.lang().global.months.may, this.$root.lang().global.months.jun, this.$root.lang().global.months.jul, this.$root.lang().global.months.aug, this.$root.lang().global.months.sep, this.$root.lang().global.months.oct, this.$root.lang().global.months.nov, this.$root.lang().global.months.dec]
+      const year = a.getFullYear()
+      const month = months[a.getMonth()]
+      const date = a.getDate().toString().length == 1 ? `0${a.getDate()}` : a.getDate()
+      return `${month} ${date}, ${year}`
```

Comment:
string date format changes per country, use moment library

Reply:
American format is dumb, we use the dd/mm/yyyy everywhere, why should we make something different depending on the country?

Reply:
Okay then, but could you do a mixin out of this code, you reuse it later and it is maybe not the last time you will